src/parser/core/modules/Weaving.tsx

```diff
@@ -59,10 +64,26 @@ export class Weaving extends Analyser {
 
 	protected severity = WEAVING_SEVERITY
 
+	//used to synth a start and end in case there were no actions before the pull or close to when the pull ended. it's also used to ensure that leading and trailing gcd events are not nullable
+	private pullStart: Events['action'] = {
+		action: 0,
```

Comment:
fyi, this parameter is fixed to be a number
0 was chosen, but I'm sure any other number could work if we don't want specifically zero